src/common/perf.py

```diff
@@ -192,6 +194,17 @@ def __init__(self, metrics_logger):
         self.all_reports = {}
         self.metrics_logger = metrics_logger
 
+    def save_to(self, perf_report_file_path=None):
+        """Saves all reports into a json file"""
+        # if no file path provided, create a temp file
+        if perf_report_file_path is None:
+            perf_report_file_path = tempfile.NamedTemporaryFile(suffix=".json").name
+
+        with open(perf_report_file_path, "w") as out_file:
+            out_file.write(json.dumps(self.all_reports, indent="    "))
```

Comment:
Why we need indent here?

Reply:
so it looks good? :) Usually when opening a json without indent it's pretty hard to read, and you'll need auto-format to make it better.

Also, having indent will make it look prettier in AzureML UI directly.